apps/frontend/app/(client)/(code-editor)/_components/ContestProblemDropdown.tsx

```diff
@@ -0,0 +1,76 @@
+'use client'
+
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger
+} from '@/components/shadcn/dropdown-menu'
+import { cn, convertToLetter, fetcherWithAuth } from '@/libs/utils'
+import checkIcon from '@/public/icons/check-green.svg'
+import type { ContestProblem, ProblemDetail } from '@/types/type'
+import { useQuery } from '@tanstack/react-query'
+import type { Route } from 'next'
+import Image from 'next/image'
+import Link from 'next/link'
+import { FaSortDown } from 'react-icons/fa'
+
+interface ContestProblemProps {
+  data: ContestProblem[]
+  total: number
+}
+
+interface ContestProblemDropdownProps {
+  problems: ContestProblemProps | undefined
+  problem: ProblemDetail
+  problemId: number
+  contestId: number
+}
+
+export default function ContestProblemDropdown({
+  problems,
+  problem,
+  problemId,
+  contestId
+}: ContestProblemDropdownProps) {
+  const { data, isSuccess } = useQuery<ContestProblemProps | undefined>({
+    queryKey: ['contest', contestId, 'problems'],
+    queryFn: () =>
+      fetcherWithAuth.get(`contest/${contestId}/problem?take=20`).json(),
+    enabled: false
+  })
+
+  if (isSuccess) problems = data
```

Comment:
```suggestion
    initialData: problems
  })
```

서버로부터 가져온 데이터를 `initialData` 옵션에 추가할 수 있습니다!

Reply:
props로 넘겨 받은 problems 랑 dropdown 컴포넌트에서 같은 API를 호출해서 enabled와 함께 지웠습니당~